components/eam/src/chemistry/mozart/mo_sethet.F90

```diff
@@ -296,109 +272,41 @@ subroutine sethet( het_rates, press, zmid,  phis, tfld, &
+    !	... Find the level index that only calculate het_rates below
     !-----------------------------------------------------------------
-    do i = 1,ncol
-       if ( abs(rlat(i)) > 60._r8*d2r ) then
-          p_limit = 300.e2_r8
-       else
-          p_limit = 100.e2_r8 
-       endif
-       k_loop: do k = pver,1,-1
-          if( press(i,k) < p_limit ) then
-             ktop(i) = k
-             exit k_loop
-          end if
-       end do k_loop
-    end do
+    call find_ktop( ncol,  rlat,  press,  & ! in
+                    ktop                  ) ! out
     ktop_all = minval( ktop(:) )
-!
-! jfl
-!
-! this is added to rescale the variable precip (which can only be positive)
-! to the actual vertical integral of positive and negative values.  This
-! removes point storms
-!
-    do i = 1,ncol
-       total_rain = 0._r8
-       total_pos  = 0._r8
-       do k = 1,pver
-          precip(i,k) = cmfdqr(i,k) + nrain(i,k) - nevapr(i,k)
-          total_rain = total_rain + precip(i,k)
-          if ( precip(i,k) < 0._r8 ) precip(i,k) = 0._r8
-          total_pos  = total_pos  + precip(i,k)
-       end do
-       if ( total_rain <= 0._r8 ) then
-          precip(i,:) = 0._r8
-       else
-          do k = 1,pver
-             precip(i,k) = precip(i,k) * total_rain/total_pos
-          end do
-       end if
-    end do
-
-    do k = 1,pver
-       !jfl       precip(:ncol,k) = cmfdqr(:ncol,k) + nrain(:ncol,k) - nevapr(:ncol,k)
-       rain(:ncol,k)   = mass_air*precip(:ncol,k)*xhnm(:ncol,k) / mass_h2o
-       xliq(:ncol,k)   = precip(:ncol,k) * delt * xhnm(:ncol,k) / avo*mass_air * m3_2_cm3
-       if( spc_hno3_ndx > 0 ) then
-          xhno3(:ncol,k)  = qin(:ncol,k,spc_hno3_ndx) * xhnm(:ncol,k)
-       else
-          xhno3(:ncol,k)  = 0._r8
-       end if
-       if( spc_h2o2_ndx > 0 ) then
-          xh2o2(:ncol,k)  = qin(:ncol,k,spc_h2o2_ndx) * xhnm(:ncol,k)
-       else
-          xh2o2(:ncol,k)  = 0._r8
-       end if
-       if( spc_sogm_ndx > 0 ) then
-          xsogm(:ncol,k)  = qin(:ncol,k,spc_sogm_ndx) * xhnm(:ncol,k)
-       else
-          xsogm(:ncol,k)  = 0._r8
-       end if
-       if( spc_sogi_ndx > 0 ) then
-          xsogi(:ncol,k)  = qin(:ncol,k,spc_sogi_ndx) * xhnm(:ncol,k)
-       else
-          xsogi(:ncol,k)  = 0._r8
-       end if
-       if( spc_sogt_ndx > 0 ) then
-          xsogt(:ncol,k)  = qin(:ncol,k,spc_sogt_ndx) * xhnm(:ncol,k)
-       else
-          xsogt(:ncol,k)  = 0._r8
-       end if
-       if( spc_sogb_ndx > 0 ) then
-          xsogb(:ncol,k)  = qin(:ncol,k,spc_sogb_ndx) * xhnm(:ncol,k)
-       else
-          xsogb(:ncol,k)  = 0._r8
-       end if
-       if( spc_sogx_ndx > 0 ) then
-          xsogx(:ncol,k)  = qin(:ncol,k,spc_sogx_ndx) * xhnm(:ncol,k)
-       else
-          xsogx(:ncol,k)  = 0._r8
-       end if
-       if( spc_so2_ndx > 0 ) then
-          xso2(:ncol,k)  = qin(:ncol,k,spc_so2_ndx) * xhnm(:ncol,k)
-       else
-          xso2(:ncol,k)  = 0._r8
-       end if
-    end do
+
```

Comment:
The `ktop_all` calculation looks problematic to me as it is taking the `minval` of the columns. The number of columns and their grouping depends on the PE layout. I am surprised that this code doesn't produce NBFB answers for different PE layouts (PET test). We can do nothing here as we like to keep the code as is.